packages/git/src/browser/git-base-widget.ts

```diff
@@ -53,4 +103,56 @@ export class GitBaseWidget extends VirtualWidget {
         }
         return result;
     }
+
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        this.addKeyListener(this.node, Key.ARROW_LEFT, () => this.handleLeft());
+        this.addKeyListener(this.node, Key.ARROW_RIGHT, () => this.handleRight());
+        this.addKeyListener(this.node, Key.ARROW_UP, () => this.handleUp());
+        this.addKeyListener(this.node, Key.ARROW_DOWN, () => this.handleDown());
+        this.addKeyListener(this.node, Key.ENTER, () => this.handleEnter());
+    }
+
+    protected handleLeft(): void {
+        this.selectNodeByDirection(SelectDirection.PREVIOUS);
+    }
+
+    protected handleRight(): void {
+        this.selectNodeByDirection(SelectDirection.NEXT);
+    }
+
+    protected handleUp(): void {
+        this.selectNodeByDirection(SelectDirection.PREVIOUS);
+    }
+
+    protected handleDown(): void {
+        this.selectNodeByDirection(SelectDirection.NEXT);
+    }
+
+    protected handleEnter(): void {
+
+    }
+
+    protected getSelected(): T | undefined {
+        return this.gitNodes ? this.gitNodes.find(c => c.selected || false) : undefined;
+    }
+
+    protected selectNode(node: T) {
+        const n = this.getSelected();
+        if (n) {
+            n.selected = false;
+        }
+        this.selectionService.selection = node;
+    }
+
+    protected selectNodeByDirection(dir: SelectDirection) {
+        const selIdx = this.gitNodes.findIndex(c => c.selected || false);
+        let nodeIdx = selIdx;
+        if (dir === SelectDirection.NEXT && selIdx < this.gitNodes.length - 1) {
+            nodeIdx = selIdx + 1;
+        } else if (dir === SelectDirection.PREVIOUS && selIdx > 0) {
+            nodeIdx = selIdx - 1;
+        }
+        this.selectNode(this.gitNodes[nodeIdx]);
```

Comment:
consider `nodeIdx` can have value of `-1` here.

Reply:
they do not, at least not for empty `this.gitNodes`.

Reply:
Ok, you're right. But in this case this method shouldn't be called at all. I will check that.